### PR TITLE
AEA-2786-Withdraw-Request-Schema

### PR DESCRIPTION
**Pull request by TomSmithCoding**

### CODE REVIEW

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        type: string
+        example: "#16708936-6397-4e03-b84f-4aaa790633e0"
+  reasonCode:
+    type: object
+    description: The reason for the task, should only be included if there is no focus or differs from the reason indicated in the focus.
+    properties:
+      coding:
+        type: array
+        items:
+          $ref: ../../components/_fragments/codes.yaml#/components/schemas/ReasonCode
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:90]: We shouldn't use a standard import here, we should be indicating the valid list of reason codes in here.

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    properties:
+      identifier:
+        $ref: ../../components/_fragments/identifiers.yaml#/components/schemas/OdsOrganizationCode
+  requester:
+    type: object
+    description: The dispenser withdrawing the prescription.
+    properties: 
+      reference:
+        type: string
+        example: "#16708936-6397-4e03-b84f-4aaa790633e0"
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:82]: This doesn't look valid, I would copy the example and description from the equivalent field in MedicationRequest.yaml

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      identifier:
+        $ref: ../../components/_fragments/identifiers.yaml#/components/schemas/Rfc4122
+  for:
+    type: object
+    properties: 
+      identifier:
+        $ref: ../../components/_fragments/identifiers.yaml#/components/schemas/NhsNumber
+  authoredOn:
+    type: string
+    description: The time the prescription is being withdrawn to Spine.
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:67]: "withdrawn *from*"

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      coding:
+        type: array
+        items:
+          $ref: ../../components/_fragments/codes.yaml#/components/schemas/TaskCode
+  groupIdentifier:
+    description: The Prescription ID of the prescription that is being withdrawn.
+    $ref: ../../components/_fragments/groupIdentifier.yaml
+  focus:
+    type: object
+    description: The release response which released the prescription that is currently being withdrawn.
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:56]: it's DN identifier, not the prescription

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    example: order
+  code: 
+    type: object
+    properties:
+      coding:
+        type: array
+        items:
+          $ref: ../../components/_fragments/codes.yaml#/components/schemas/TaskCode
+  groupIdentifier:
+    description: The Prescription ID of the prescription that is being withdrawn.
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:52]: "The prescription id that the DN was originally submitted against.

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    properties:
+      coding: 
+        type: array
+        items:
+          $ref: ../../components/_fragments/codes.yaml#/components/schemas/EpsTaskDispenseWithdrawStatusReason
+  intent:
+    type: string
+    description: The intent of the Medication Requests being withdrawn.
+    example: order
+  code:
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:44]: Not sure what this is, can't see it being used in our code.

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    items:
+      $ref: ../../components/_extensions/EPSRepeatInformation.yaml
+  identifier:
+    description: The unique identifier for the Withdraw Request.
+    type: array
+    items:
+      $ref: ../../components/_fragments/identifiers.yaml#/components/schemas/Rfc4122
+  status: 
+    type: string
+    example: in-progress
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:31]: this is the only value that's valid for a withdraw not just an example, so should be enum

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      oneOf:
+      - $ref: ../../components/PractitionerRole/TaskPractitionerRole.yaml
+      - $ref: ../../components/Organization/DispenseNotificationOrganization.yaml
+  extension:
+    type: array
+    description: |
+      FHIR extension array to hold Repeat Information.
+      This identifies the instance of the eRD prescription that is being returned.
+      Given that all eRD prescriptions have the same prescription ID,
+      the combination of Prescription ID and repeat number identifies the issue to withdraw or return.
```
Comment (MatthewPopat-NHS) [inline packages/specification/schemas/requests/dispensing/Withdraw.yaml:21]: "... identifies the issue *that the DN was originally submitted against.*"

File: packages/specification/schemas/requests/dispensing/Withdraw.yaml (hunk 6 above)
Comment (TomSmithCoding) [reply packages/specification/schemas/requests/dispensing/Withdraw.yaml:44]: Task-code I can currently see being used in task-validator.ts and translation.ts, unless you're referring to the Intent which I can only see being forced to TaskIntent.ORDER in translation, and anything else fails the task-validator

Comment (MatthewPopat-NHS) [reply packages/specification/schemas/requests/dispensing/Withdraw.yaml:55]: Ah yes, again we should probably inline this and explicitly state that it must be "abort".

Comment (TomSmithCoding) [reply packages/specification/schemas/requests/dispensing/Withdraw.yaml:55]: Will do

**Review by originalphil — APPROVED** [review]

**Pull request merged**